cns/restserver/api.go

```diff
@@ -1139,19 +1139,21 @@ func (h *HTTPRestService) doPublish(ctx context.Context, req cns.PublishNetworkC
 	var innerReq nmagent.PutNetworkContainerRequest
 	err := json.Unmarshal(innerReqBytes, &innerReq)
 	if err != nil {
-		returnMessage := fmt.Sprintf("Failed to publish Network Container: %s", req.NetworkContainerID)
+		returnMessage := fmt.Sprintf("Failed to unmarshal embedded NC publish request for NC %s, with err: %v", req.NetworkContainerID, err)
 		returnCode := types.NetworkContainerPublishFailed
 		logger.Errorf("[Azure-CNS] %s", returnMessage)
 		return returnMessage, returnCode
 	}
 
 	innerReq.AuthenticationToken = ncParameters.AuthToken
 	innerReq.PrimaryAddress = ncParameters.AssociatedInterfaceID
+	innerReq.ID = req.NetworkContainerID
 
+	logger.Printf("innerReq right before nma call %+v", innerReq)
 	err = h.nma.PutNetworkContainer(ctx, &innerReq)
 	// nolint:bodyclose // existing code needs refactoring
 	if err != nil {
-		returnMessage := fmt.Sprintf("Failed to publish Network Container: %s", req.NetworkContainerID)
+		returnMessage := fmt.Sprintf("Failed to publish Network Container in put Network Conatainer call: %s with err: %v", req.NetworkContainerID, err)
```

Comment:
```suggestion
		returnMessage := fmt.Sprintf("Failed to publish Network Container %s in put Network Container call, with err: %v", req.NetworkContainerID, err)
```
makes the message read better and fixes the type